### ISSUE TITLE
Packer gives misleading errors when IAM permissions are missing

### ISSUE THREAD

**Issue by codesurf42** [hashicorp/packer#6469]
This is continuation of https://github.com/hashicorp/packer/issues/4277
with recent packer version as the problem still occurs.

packer 1.2.4
Aws ec2

When ec2 instance doesn't have enough IAM permissions, they are reported in a way that doesn't tell you what you should change. This may be coming from go aws sdk, so there may not be an easy way to fix it - which would be to report specific AWS IAM permission names that are missing to perform specific operation.

```
==> amazon-ebs: Modifying attributes on AMI (ami-67e80000)...
    amazon-ebs: Modifying: users
==> amazon-ebs: Error modify AMI attributes: UnauthorizedOperation: You are not authorized to perform this operation.
==> amazon-ebs: 	status code: 403, request id: 032c86ae-2f73-4140-8fca-9207eb96f0d6
==> amazon-ebs: Deregistering the AMI because cancellation or error...
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: Error describing volumes: UnauthorizedOperation: You are not authorized to perform this operation.
==> amazon-ebs: 	status code: 403, request id: 56f87547-f8fd-49e6-8d2e-c7c57f5ce05f
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' errored: Error modify AMI attributes: UnauthorizedOperation: You are not authorized to perform this operation.
	status code: 403, request id
```

How to repeat?
Run `packer build` on ec2 instance with not enough permissions, e.g. less than described in the link below.

For reference, the minimal working set of permissions is listed on https://www.packer.io/docs/builders/amazon.html

**Comment by rickard-von-essen** [comment]
Which permission was missing?

**Comment by codesurf42** [comment]
I don't know exactly which one (as you can see from the attached logs), there were few of them missing regarding ECR and after merging current IAM role with what is listed on linked page, it went ok. It is not an issue for me anymore, just a general problem with how packer is reporting such missing permissions.

**Comment by SwampDragons** [comment]
Yeah, it looks like we're directly surfacing the error provided by the AWS SDK so I'm not sure of an easy way to surface this information.

**Comment by rickard-von-essen** [comment]
I think there are two types the ones where we hide the error, usually when we discard the error in a waiting loop. That's a bug.

But if the error is like:

```
 Modifying attributes on AMI (ami-67e80000)...
    amazon-ebs: Modifying: users
==> amazon-ebs: Error modify AMI attributes: UnauthorizedOperation: You are not authorized to perform this operation.
==> amazon-ebs: 	status code: 403, request id: 032c86ae-2f73-4140-8fca-9207eb96f0d6
```

Because the user is missing `ec2:ModifyImageAttribute` permissions then that should be fairly obvious for the user.

Hopefully we have weeded out most of the first type.

**Comment by codesurf42** [comment]
It was really quick turn-around and very encouraging to contribute more. Thanks!
